drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -622,19 +659,50 @@ static int mtd_spi_nor_write_page(mtd_dev_t *mtd, const void *src, uint32_t page
     /* write enable */
     mtd_spi_cmd(dev, dev->params->opcode->wren);
 
+    /* Wait for WEL-Flag to be set */
+    wait_for_write_enable_set(dev);
+
     /* Page program */
     mtd_spi_cmd_addr_write(dev, dev->params->opcode->page_program, addr, src, size);
 
     /* waiting for the command to complete before returning */
     wait_for_write_complete(dev, 0);
 
+#if IS_USED(MODULE_MTD_SPI_NOR_MX_SECURITY)
+    /* Read security register */
+    uint8_t scur_reg;
+    mtd_spi_cmd_read(dev, dev->params->opcode->rdscur, &scur_reg, sizeof(scur_reg));
+    if (scur_reg & MX_SECURITY_PFAIL) {
+        DEBUG("mtd_spi_nor_write: ERR: page program failed! scur = %02x\n", scur_reg);
+        ret = -EIO;
+    }
+#elif IS_USED(MODULE_MTD_SPI_NOR_ISSI_SECURITY)
+    /* Read extended read register for security flags */
+    uint8_t erp_reg;
+    mtd_spi_cmd_read(dev, dev->params->opcode->rderp, &erp_reg, sizeof(erp_reg));
+    if (erp_reg & IS_ERP_P_ERR) {
+        DEBUG("mtd_spi_nor_write: ERR: page program failed! erp = %02x\n", erp_reg);
+        ret = -EIO;
+
+        /* clear the extended read parameters register manually */
+        mtd_spi_cmd(dev, dev->params->opcode->clerp);
+
+        /* ISSI flashs do not reset the WEL-Flag when an operation was not *completed* */
+        mtd_spi_cmd(dev, dev->params->opcode->wrdi);
+    }
+#endif
```

Comment:
better move this block to a helper function to keep the code tidy.

Reply:
Will do :)